### ISSUE TITLE
Module ... has no attribute ... when #type: ignore[misc] is set at the top of a module

### ISSUE THREAD

**Issue by johanvergeer** [python/mypy#9318]
In our mypy config file we do not allow `Any` to be used. Because this cannot be avoided all the time we use `#type: ignore[misc]` at the top of a module. 

We created a module called `typing.py`. 

```python
# type: ignore[misc] # Ignores 'Any' input parameter
from typing import Any, Iterable

JsonSerializable = Any
CsvSerializable = Iterable[Any]
```

This is imported in another module.

```python
from my_app.typing import JsonSerializable

def foo(data: JsonSerializable) -> None:
    ...
```

# Actual result

```powershell
PS C:\workspace\my_app> mypy
src\my_app\foo.py:6: error: Module 'my_app.typing' has no attribute 'JsonSerializable'  [attr-defined]
    from my_app.typing import JsonSerializable

Found 1 errors in 1 files (checked 6 source files)
```

# Expected result

```powershell
PS C:\workspace\my_app> mypy
Success: no issues found in 6 source files
```

# How did we solve it?

When we change `typing.py` so it ignores `Any` on each line we don't have any issues.

```python
from typing import Any, Iterable

JsonSerializable = Any  # type: ignore[misc] # Ignores 'Any' input parameter
CsvSerializable = Iterable[Any]  # type: ignore[misc] # Ignores 'Any' input parameter
```

# Configuration

- Python 3.8.2
- Mypy 0.782
- Windows 10
- Not in a virtualenv

## .mypy.ini

```ini
[mypy]
python_version = 3.8

disallow_any_expr = False
disallow_any_decorated = False
disallow_any_explicit = True
disallow_any_generics = True
disallow_subclassing_any = True

disallow_untyped_calls = True
disallow_untyped_defs = True
check_untyped_defs = False
disallow_untyped_decorators = False

no_implicit_optional = True

warn_redundant_casts = True
warn_unused_ignores = True
warn_return_any = True
warn_unreachable = True
strict_equality = True

allow_redefinition = False
```

**Comment by ilevkivskyi** [comment]
Yeah, this looks like a bug.

**Comment by RomainMuller** [comment]
I've got the exact same issue happening with file-wide `# typing: ignore[no-any-return]`, so I also believe this isn't quite specific to ignoring any particular code; instead related to file-wide ignoring *anything*.

**Comment by MetRonnie** [comment]
I had the same error message when importing from an auto-generated protobuf `_pb2.py` file, which has `# type: ignore` as the first line. I managed to suppress the errors by adding `# type: ignore` to the end of every line where we imported from the protobuf file, but that's pretty annoying.

**Comment by aberres** [comment]
Yeps, same problem here.

**Comment by dpinol** [comment]
The workaround at the description is a pain when "type: ignore" needs to be placed in hundreds of lines. 
Is there any other workaround?

**Comment by hauntsaninja** [comment]
Maybe a per module https://mypy.readthedocs.io/en/stable/config_file.html#confval-ignore_errors works for your use case?
